src/components/relatedProducts/comparison/FeaturesList.jsx

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable */
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+
+const FeaturesList = ({ features, relatedFeatures }) => {
+  const [allFeatures, setAllFeatures] = useState([]);
+  const [uniq, setUniq] = useState([]);
+  const [total, setTotal] = useState({});
+  const featNames = [];
+  const shared = {};
+
+  const unique = () => {
+    allFeatures.forEach(feature => featNames.push(feature.feature))
+  }
+
+  const buildShared = () => {
+    features.forEach(feature => {
+      shared[feature.feature] = [feature.value, null];
+    })
+    relatedFeatures.forEach(feature => {
+      if (shared[feature.feature]) {
+        shared[feature.feature][1] = feature.value;
+      } else {
+        shared[feature.feature] = [null, feature.value];
+      }
+    })
+    setTotal(shared);
+  }
+
+  useEffect(() => {
+    buildShared()
+  }, [features])
+
+  useEffect(() => {
+    setAllFeatures([...features, ...relatedFeatures])
+  }, [features])
+
+  useEffect(() => {
```

Comment:
I wonder if you could combine the use effect on this line with the use effect on the line starting at 31. 
It could look like this:
useEffect(() => {
   buildShared();
   setAllFeatures([...features, ...relatedFeatures])l
}, [features])

That would save you a few lines of code.